uint32[] joints
float64[] joint_thetas